test/sequential/rolldown.withdraw.fees.test.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ * @group withdrawal-rolldown
+ */
+
+import { ApiPromise } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { ETH_ASSET_ID, GASP_ASSET_ID } from "../../utils/Constants";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { User } from "../../utils/User";
+import { Sudo } from "../../utils/sudo";
+import { Assets } from "../../utils/Assets";
+import { BN_TEN, BN_THOUSAND, signTx } from "gasp-sdk";
+import { Withdraw } from "../../utils/rolldown";
+import { BN, BN_MILLION } from "@polkadot/util";
+import {
+  ExtrinsicResult,
+  filterZeroEventData,
+} from "../../utils/eventListeners";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+
+let api: ApiPromise;
+let testUser: User;
+let gaspIdL1Asset: any;
+let withdrawalAmount: BN;
+let treasuryAccount: string;
+let ethIdL1Asset: any;
+let DEFAULT_AMOUNT: any;
+let chain: any;
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  await setupApi();
+  api = getApi();
+  gaspIdL1Asset = JSON.parse(
+    JSON.stringify(await api.query.assetRegistry.idToL1Asset(GASP_ASSET_ID)),
+  );
+  ethIdL1Asset = JSON.parse(
+    JSON.stringify(await api.query.assetRegistry.idToL1Asset(ETH_ASSET_ID)),
+  );
+  withdrawalAmount = BN_MILLION;
+  treasuryAccount = "0x6d6f646c70792f74727372790000000000000000";
+  DEFAULT_AMOUNT = BN_THOUSAND.mul(BN_TEN.pow(new BN(18)));
+  chain = "Ethereum";
+});
+
+beforeEach(async () => {
+  [testUser] = setupUsers();
+});
+
+test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing GASP, some fees goes to treasury", async () => {
+  const treasuryBalanceBefore = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+  const events = await signTx(
+    getApi(),
+    await Withdraw(testUser, withdrawalAmount, gaspIdL1Asset.ethereum, chain),
+    testUser.keyRingPair,
+  );
+  const eventFiltered = filterZeroEventData(events, "Transfer");
```

Comment:
What does "filterZeroEventData" do? :O

Reply:
Convert all data from the filtered event to "normal" format (without things like .toHuman.data[0][4])

Reply:
then rename it like , filterAndGetFirstEventWithString , or something like that plz,

Reply:
Done